webapp/components/news-list/NewsList.vue

```diff
@@ -31,32 +31,84 @@
 
 import NewsItem from '@/components/news-item/NewsItem';
 import NewsForm from '@/components/news-form/NewsForm';
+import gql from 'graphql-tag';
+import { mapGetters } from 'vuex'
 
 export default {
   name: 'NewsList',
   components: { NewsItem, NewsForm },
-  props: {
-    newsItems: {type: Array, default() { return [] } },
-    descending: {type: Boolean, default: true},
+
+  data: () => {
+    return {
+      newsItems: [],
+      descending: true,
+    }
   },
   methods: {
     updateNews(newsToUpdate) {
       let current = this.newsItems.findIndex(newsItem => newsItem.title === newsToUpdate.title);
       this.newsItems.splice(current, 1, newsToUpdate);
     },
-    removeNews(newsToRemove) {
-      this.newsItems = this.newsItems.filter(item => item.title !== newsToRemove.title);
+    async removeNews(newsToRemove) {
+      const mutation = gql`
+        mutation($title: ID!) {
+          delete(title: $title) {
+            title
+          }
+        }
+      `;
+      try {
+        const { data } = await this.$apollo.mutate({ mutation, variables: { title: newsToRemove.title }});
+        const { title } = data.delete;
+        this.newsItems = this.newsItems.filter(item => item.title !== title);
+      }
+      catch (e) { console.log(e.message); }
     },
-    addItem(title) {
-      this.newsItems.push({ title: title, votes: 0 });
+    async addItem(title) {
+      const mutation = gql`
+        mutation($post: PostInput!) {
+            write(post: $post) {
+                title,
+                votes,
+                author {
+                  id
+                }
+            }
+        }
+      `;
+      const variables = { post: { title } };
+      try {
+          const { data: { write } } = await this.$apollo.mutate({ mutation, variables })
+          this.newsItems.push({ title: write.title, votes: write.votes, author: write.author});
+      }
+      catch (e) { console.log(e.message) }
     },
     toggleSortOrder() {
       this.descending = !this.descending;
     },
+    async getNews() {
+      const query = gql`
+        query {
+            posts {
+                title
+                votes
+                author {
+                  id
+                }
+            }
+        }
+      `;
+      try {
+        const { data: { posts }} = await this.$apollo.query( { query } );
+        return posts;
+      }
+      catch (e) {
+        return [];
+      }
+    }
```

Comment:
```suggestion
apollo: {
  posts: gql`
        query {
            posts {
                title
                votes
                author {
                  id
                }
            }
        }
      `
}
```


Will automatically populate `this.posts` for you! :muscle: 


https://apollo.vuejs.org/guide/apollo/#queries